### ISSUE TITLE
Microsoft Teams: @name turns to regular text (no actual link to people or notification of them) as soon as one hits spacebar

### ISSUE THREAD

**Issue by georgeogoodman** [wavebox/waveboxapp#804]
As in the Subject:
1. start a new message into a Channel on Microsoft Teams
2. use @name to flag a person and select the desired match from the pop-up
3. hit spacebar to continue the message and note that the @name has become regular text. Also, once the message is sent, the person/people flagged with @name-style tags do not get notified that they were mentioned.

* Wavebox Version: [e.g. 3.0.0]
* Operating System & Version: [e.g. Windows 10, Mac OSX 10.3.4, Ubuntu 17.10]
* (Linux only) Install method: [e.g. Debian installer, Snap Package, Tar, Apt-Repo, etc...]
* Account type (if applicable): [e.g. Gmail, Office 365, Trello, etc...]

**Comment by Thomas101** [comment]
Hi, thanks for reporting. I think this is the desired behaviour that Microsoft has added to teams - I see the same behaviour in the browser too.

There are a few ways to link the name in though...
1. Type the complete username (e.g. `@Thomas101` works but `@Thoma` does).
2. Type part of the username and then press tab to complete it (e.g. `@Thoma` + <kbd>Tab</kbd> will complete it to `@Thomas101`)
3. Type part of the username and then click one of the users in the suggested list

Have you come across the `@Thoma` + <kbd>space</kbd> behaviour when using Teams anywhere else (e.g. mobile)?

**Comment by georgeogoodman** [comment]
Thanks @Thomas101 for the ideas. 
When I type the complete @username, it "lights up" as one would hope. Clicking the item in the dropdown leaves the text looking like it is correct. If I then continue to type the rest of a message, the @username text returns to being regular text and the user is not notified once I post the message.
Likewise, if I do as you suggest in your #2 suggestion, the moment I continue with the message the @username text returns to being regular text and the user is not notified once I post the message.
Sadly, this is also true for your #3 suggestion.

The only way I've found that I can successfully include an actual @username reference in a message on Microsoft Teams when accessing that service through Wavebox is if I type the entire message, then go back to where I want the @username reference, and enter the @username reference...and then post the message without typing anything additional.

This is not the behavior I see using Chrome on ChromeOS, Chrome on macOS, Chrome on Debian Linux, Chrome on Windows, or the desktop clients for Teams on Windows or macOS.

**Comment by georgeogoodman** [comment]
OK, this is weird (at least to me). I have now verified that there seems to be an interaction with the custom Agent string I am using (in order to actually get video and audio chat working for Teams on Linux). If I run teams in Wavebox with the default Agent string, the @username stuff all behaves as one would expect. If, on the other hand, I apply the custom Agent string (Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/19.10136) in the Microsoft Teams account settings (the Advanced section) of Wavebox, then the @username support does all the odd things I've documented earlier in this thread.

So, I need to choose between audio/video support and convenient use of @username. For now, I'll go with audio/video but would love to have both work.

Thanks for the assist in isolating this.
